File: backend/src/main/java/com/example/entities/MonthlyPlan.java
package com.example.entities;

import java.time.LocalDate;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "monthly_plan")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class MonthlyPlan {
	
	@Id
	@Column(name="plan_id")
	int planId;
	
	@Column(name="plan_name")
	String planName;
	
	@Column(name="monthly_price")
	double monthlyPrice;
	
	@Column(name="meal_inclusion")
	String mealInclusion;         //Lunch, Dinner, Both
	
	@Column(name="validity_period")
	int validityPeriod;               // number of days the plan will be valid
	
	@ManyToOne
	@JoinColumn(name = "mess_id")
	private Mess mess;
	
	

}

File: backend/Customer/Customer/Models/MonthlyPlan.cs
using System;
using System.Collections.Generic;

namespace Customer.Models;

public partial class MonthlyPlan
{
    public int PlanId { get; set; }

    public int MessId { get; set; }

    public string? PlanName { get; set; }

    public decimal MonthlyPrice { get; set; }

    public string MealInclusion { get; set; } = null!;

    public int ValidityPeriod { get; set; }

    public virtual Mess Mess { get; set; } = null!;

    public virtual ICollection<Subscription> Subscriptions { get; set; } = new List<Subscription>();
}
